### PR TITLE
Edit open-source software benefits for research

### LINKED ISSUE #1183: toolboxes
## Detailed description
This issue is a place to discuss matters relating to the writing of the chapter on the importance of open-source toolboxes for reproducibility. 

### Table of contents
A few ideas:

* toolboxes help ensure that key parts of research code are co-developed, and unit and field tested by the wider research community
* they facilitate transparency and checkability of research code (clear abstractions, common interfaces, etc.)
* they promote appropriate workflows
* they facilitate readability

### Resources
I'm sure people have written about his before, but I'm not aware of any reference. 

## Current status

This chapter is empty. If anyone would like to make a start they are more than welcome to do so.

 - [ ] Write chapter outline
 - [ ] Add material to the chapter
 - [ ] Combine materials into a readable chapter
 - [ ] Proofread
 - [ ] Request reviews
 - [ ] Address reviews
 - [ ] Merge to master branch.

### PR DESCRIPTION
**Pull request by mloning**
### Summary

* Restructures section "How open-source software benefits research"

### List of changes proposed in this PR (pull-request)

* structures the section by bullet points and adjectives to summarise bullet points
* adds idea that open source software is checkable by others

### Comments
* I opened a separate PR for the chapter on toolboxes

### CODE REVIEW

File: book/website/reproducible-research/open/open-source.md [hunk 1]
```diff
@@ -52,15 +52,12 @@ Open source provides an endless, ever-changing set of Rubix cubes for you to sol
...
+* Checkable. Open-source projects allow the wider research community to read and test each others' code. This way, bugs can be found more quickly and results can be validated by other researchers. 
+* Collaborative. Working openly also allows any number of researchers to collaborate on projects that could not possibly be developed by single researchers/research groups. Examples include [Linux](https://www.linux.org/) operating systems, Python packages such as [scipy](https://www.scipy.org/) and [numpy](http://www.numpy.org/), and the machine learning library [TensorFlow](https://www.tensorflow.org/).
```
Comment (malvikasharan) [inline book/website/reproducible-research/open/open-source.md:59]:
```suggestion
#### Re-usable

Open source software projects allow researchers to take advantage of each others’ work. This enables researchers to apply their efforts to high-value work. 
It is sometimes said that “all the easy problems have already been solved”. 
Blogging, content management, and operating systems are all problems with established (and mainstream) open source solutions, to name a few. 
While developers could spend their time reinventing wheels that the open source community have already perfected, it is highly preferable to use the world’s best wheel, especially when that wheel comes at no cost to you. 
This reduces duplication of effort and allows researchers to focus on yet-unsolved challenges.

#### Checkable

Open-source projects allow the wider research community to read and test each others' code. This way, bugs can be found more quickly and results can be validated by other researchers. 

#### Collaborative
Working openly also allows any number of researchers to collaborate on projects that could not possibly be developed by single researchers/research groups. 
Examples include [Linux](https://www.linux.org/) operating systems, Python packages such as [scipy](https://www.scipy.org/) and [numpy](http://www.numpy.org/), and the machine learning library [TensorFlow](https://www.tensorflow.org/).
```

**Comment by malvikasharan** [comment]
Thank @mloning for adding a new section. I have done the minor formatting as per the style guide to make it easy to read and review in future.

@all-contributors please add @mloning for content.

**Comment by malvikasharan** [comment]
Please let me know if it's ready to merge.

**Comment by mloning** [comment]
Thanks @malvikasharan for the review! It's good to go from my side.

File: book/website/reproducible-research/open/open-source.md [hunk 2]
```diff
@@ -54,9 +54,21 @@ Open source provides an endless, ever-changing set of Rubix cubes for you to sol
...
-* Checkable. Open-source projects allow the wider research community to read and test each others' code. This way, bugs can be found more quickly and results can be validated by other researchers. 
-* Collaborative. Working openly also allows any number of researchers to collaborate on projects that could not possibly be developed by single researchers/research groups. Examples include [Linux](https://www.linux.org/) operating systems, Python packages such as [scipy](https://www.scipy.org/) and [numpy](http://www.numpy.org/), and the machine learning library [TensorFlow](https://www.tensorflow.org/).
+#### Re-usable
+
+Open source software projects allow researchers to take advantage of each others’ work. This enables researchers to apply their efforts to high-value work.
```
Comment (malvikasharan) [inline book/website/reproducible-research/open/open-source.md:59]:
```suggestion
Open source software projects allow researchers to take advantage of each others’ work. 
This enables researchers to apply their efforts to high-value work. 
```

File: book/website/reproducible-research/open/open-source.md [hunk 3]
```diff
@@ -54,9 +54,21 @@ Open source provides an endless, ever-changing set of Rubix cubes for you to sol
...
-* Collaborative. Working openly also allows any number of researchers to collaborate on projects that could not possibly be developed by single researchers/research groups. Examples include [Linux](https://www.linux.org/) operating systems, Python packages such as [scipy](https://www.scipy.org/) and [numpy](http://www.numpy.org/), and the machine learning library [TensorFlow](https://www.tensorflow.org/).
+#### Re-usable
+
+Open source software projects allow researchers to take advantage of each others’ work. This enables researchers to apply their efforts to high-value work. 
+It is sometimes said that “all the easy problems have already been solved”. 
+Blogging, content management, and operating systems are all problems with established (and mainstream) open source solutions, to name a few. 
+While developers could spend their time reinventing wheels that the open source community have already perfected, it is highly preferable to use the world’s best wheel, especially when that wheel comes at no cost to you. 
+This reduces duplication of effort and allows researchers to focus on yet-unsolved challenges.
```
Comment (malvikasharan) [inline book/website/reproducible-research/open/open-source.md:63]:
```suggestion
It is sometimes said that “all the easy problems have already been solved”. 
Blogging, content management, and operating systems are all problems with established (and mainstream) open source solutions, to name a few. 
While developers could spend their time reinventing wheels that the open source community has already perfected, it is highly preferable to use the world’s best wheel, especially when that wheel comes at no cost to you. 
This reduces duplication of effort and allows researchers to focus on yet-unsolved challenges.
```

File: book/website/reproducible-research/open/open-source.md [hunk 4]
```diff
@@ -54,9 +54,21 @@ Open source provides an endless, ever-changing set of Rubix cubes for you to sol
...
+
+#### Checkable
+
+Open-source projects allow the wider research community to read and test each others' code. This way, bugs can be found more quickly and results can be validated by other researchers.
```
Comment (malvikasharan) [inline book/website/reproducible-research/open/open-source.md:67]:
```suggestion
Open-source projects allow the wider research community to read and test each others' code. 
This way, bugs can be found more quickly and results can be validated by other researchers. 
```

**Pull request merged**